File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/AccountingModule/Bank.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.AccountingModule
{
    public class Bank : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name cannot be longer than 100 characters.")]
        public string Name { get; set; }
        public string BankAccountName { get; set; }
        public string BankName { get; set; }
        public string IBAN { get; set; }
        public int BankAccountNumber { get; set; }
        public string AccountOpeningDate { get; set; }
        public string AccountClosingDate { get; set; }

        [ForeignKey("AccountNumber")]
        public long AccountNumberID { get; set; }

        public AccountingTreeChart AccountNumber { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/Accounting/bank.ts
export class Bank {
    constructor(
        public id: number = 0,
        public name: string = '',
        public bankAccountName: string = '',
        public bankName: string = '',
        public iban: string = '',
        public bankAccountNumber: number = 0,
        public accountOpeningDate: string = '',
        public accountClosingDate: string = '',
        public accountNumberName: string = '',
        public accountNumberId :number =0,
        public insertedByUserId :number =0,
        public insertedAt :number =0,
    ) {}
}

